### DISCUSSION TITLE
Enhancement Request: Data Import to support concurent connections

### DISCUSSION THREAD

**Discussion by Joseph-Duty-VA** [microsoft/powerplatform-actions#474] [Ideas]
With the latest version of PAC CLI, there appears to be support for parallel thread processing: ( https://learn.microsoft.com/en-us/power-platform/developer/cli/reference/data#--connection-count--c )

Surfacing this as an optional parameter through GitHub actions would be a significant improvement as the current approach runs on a single thread, making it process very slowly for larger datasets.  (alternately, defaulting to a higher number than 1 would also be helpful).

**Comment by devkeydet** [comment]
Will look at exposing the `--connection-count` parameter.  However, note that the default is already 5.
![image](https://github.com/microsoft/powerplatform-actions/assets/4499296/f9546df0-cb57-437a-a368-22bf578e87b8)

Therefore, increasing it may not give you the performance gain you are expecting.  `pac data import` uses the `CMT` engine.
https://learn.microsoft.com/en-us/power-platform/admin/manage-configuration-data

It has built in intelligence of data relationships.  Therefore, it can only import data concurrently if data relationships allow for it.  In many cases the data that something refers to has to be imported first before the data referring to it can be imported.  This would be the case if you were manually importing data with the underlying APIs `pac data import` uses.

**Reply by devkeydet** [reply]
PR for this enhancement request was merged.
https://github.com/microsoft/powerplatform-actions/pull/566

Stay tuned for the next release to include this enhancement.
